serve.go

```diff
@@ -18,22 +16,30 @@ import (
 
 func Serve(c *util.Config) {
 	endpoint := c.Endpoint
+	health, err := newHealthHandler(c)
+	if err != nil {
+		log.Fatalf("Error: %v", err)
+	}
+	http.Handle(endpoint, health)
+	log.Printf("Starting to listen on: %s", c.ListenAddress)
+	log.Fatal(http.ListenAndServe(c.ListenAddress, nil))
+}
+
+func newHealthHandler(c *util.Config) (*healthHandler, error) {
 	color.NoColor = true
 	cache := cache.New(c.Cache, 30*time.Second)
 
 	cfg, err := getGossConfig(c.Vars, c.VarsInline, c.Spec)
 	if err != nil {
-		fmt.Printf("Error: %v\n", err)
-		os.Exit(1)
+		return nil, err
 	}
 
 	output, err := getOutputer(c.NoColor, c.OutputFormat)
 	if err != nil {
-		fmt.Printf("Error: %v\n", err)
-		os.Exit(1)
```

Comment:
I replaced these exits with the `log.Fatalf` inside the sub-command, since it makes this easier to write coverage for at some stage. If the system-level exits remain, that would kill the test-runner if it went down this branch.